Blockzilla/SettingsViewController.swift

```diff
@@ -327,28 +333,30 @@ class SettingsViewController: UIViewController, UITableViewDataSource, UITableVi
         let subtitle: String
         
         switch biometryType {
-            case .faceID:
-                label = UIConstants.strings.labelFaceIDLogin
-                subtitle = String(format: UIConstants.strings.labelFaceIDLoginDescription, AppInfo.productName)
-            case .touchID:
-                label = UIConstants.strings.labelTouchIDLogin
-                subtitle = String(format: UIConstants.strings.labelTouchIDLoginDescription, AppInfo.productName)
-            default:
-                // Unknown biometric type
-                return nil
+        case .faceID:
+            label = UIConstants.strings.labelFaceIDLogin
+            subtitle = String(format: UIConstants.strings.labelFaceIDLoginDescription, AppInfo.productName)
+        case .touchID:
+            label = UIConstants.strings.labelTouchIDLogin
+            subtitle = String(format: UIConstants.strings.labelTouchIDLoginDescription, AppInfo.productName)
+        default:
+            // Unknown biometric type
+            return nil
         }
         
         let toggle = BlockerToggle(label: label, setting: SettingsToggle.biometricLogin, subtitle: subtitle)
         toggle.toggle.isEnabled = biometryType.hasIdentities
         return toggle
     }
-
-    fileprivate func toggleForIndexPath(_ indexPath: IndexPath) -> BlockerToggle {
-        var index = (indexPath as NSIndexPath).row
-        for i in 0..<(indexPath as NSIndexPath).section {
-            index += tableView.numberOfRows(inSection: i)
-        }
-        guard let toggle = toggles[index] else { return BlockerToggle(label: "Error", setting: SettingsToggle.blockAds) }
+    
+    fileprivate func toggleForIndexPath(_ indexPath: IndexPath) -> BlockerToggle? {
+        // Necessary check for func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath)
+        let sec: Int = (indexPath as NSIndexPath).section
+        let row: Int = (indexPath as NSIndexPath).row
+        if ((sec + 1) > toggles.count) || ((row + 1) > toggles[sec].count) {
+            return nil
+        }
+        guard let toggle = toggles[sec][row] else { return BlockerToggle(label: "Error", setting: SettingsToggle.blockAds) }
```

Comment:
We don't return a fatal error because this is only called with an incorrect index from the function that is just getting a generic toggle cell height. It's either this, or change this to return an optional toggle, and have an "if let {} else{}" in the getCell height function.